Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Route.hs

```diff
@@ -21,17 +21,52 @@ module Domain.Action.UI.Route
   )
 where
 
+import Data.Geohash as DG
+import Data.List.NonEmpty as NE
+import Data.Text (pack)
+import Data.Time as DT
+  ( TimeZone (TimeZone),
+    defaultTimeLocale,
+    formatTime,
+    getCurrentTime,
+    utcToLocalTime,
+  )
+import Domain.Types.Maps.DirectionsCache as DC
 import qualified Domain.Types.Merchant as Merchant
 import qualified Domain.Types.Person as DP
+import Kernel.Prelude
+import qualified Kernel.Storage.Esqueleto as Esq
+import Kernel.Types.Error (MerchantError (MerchantNotFound))
 import Kernel.Types.Id
 import Kernel.Utils.Common
 import Storage.CachedQueries.CacheConfig (CacheFlow)
+import Storage.CachedQueries.Maps.DirectionsCache as DCC
+import qualified Storage.CachedQueries.Maps.DirectionsCache as DQ
+import qualified Storage.CachedQueries.Merchant as QMerchant
 import qualified Tools.Maps as Maps
 import Tools.Metrics (CoreMetrics)
 
 getRoutes :: (EncFlow m r, CacheFlow m r, EsqDBFlow m r, CoreMetrics m) => (Id DP.Person, Id Merchant.Merchant) -> Maps.GetRoutesReq -> m Maps.GetRoutesResp
 getRoutes (_, merchantId) req = do
-  Maps.getRoutes merchantId req
+  merchant <- QMerchant.findById merchantId >>= fromMaybeM (MerchantNotFound merchantId.getId)
+  let origin = NE.head req.waypoints
+  let dest = NE.last req.waypoints
+  let originGeoHash = pack $ fromMaybe "Invalid origin hash" $ DG.encode merchant.geoHashPrecisionValue (origin.lat, origin.lon)
+  let destGeoHash = pack $ fromMaybe "Invalid dest hash" $ DG.encode merchant.geoHashPrecisionValue (dest.lat, dest.lon)
```

Comment:
Do not use `fromMaybe`, throw an exception if there is a problem calculating the hash